File: java/src/main/java/com/workshop/architecture/fitness/business/application/ActivateMembershipResult.java
package com.workshop.architecture.fitness.business.application;

import java.time.LocalDate;

public record ActivateMembershipResult(
        String membershipId,
        String customerId,
        String planId,
        int planPrice,
        int planDuration,
        String status,
        LocalDate startDate,
        LocalDate endDate,
        String invoiceId,
        String externalInvoiceId,
        LocalDate invoiceDueDate
) {
}

File: python/src/workshop_api/fitness/business/application/activate_membership_result.py
from dataclasses import dataclass
from datetime import date
from uuid import UUID


@dataclass(frozen=True)
class ActivateMembershipResult:
    membership_id: UUID
    customer_id: UUID
    plan_id: UUID
    plan_price: int
    plan_duration: int
    status: str
    start_date: date
    end_date: date
    invoice_id: UUID
    external_invoice_id: str | None
    invoice_due_date: date
